### PR TITLE
python311Packages.ibis-framework: 7.1.0 -> 8.0.0

### PR DESCRIPTION
**Pull request by dotlambda**
## Description of changes
Diff: https://github.com/ibis-project/ibis/compare/refs/tags/7.1.0...8.0.0

Changelog: https://github.com/ibis-project/ibis/blob/8.0.0/docs/release_notes.md

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

closes https://github.com/NixOS/nixpkgs/pull/288520

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by a-n-n-a-l-e-e** [comment]
- see also https://github.com/NixOS/nixpkgs/pull/288207

**Comment by a-n-n-a-l-e-e** [comment]
> * see also [python311Packages.ibis-framework: 7.1.0 -> 8.0.0 #288207](https://github.com/NixOS/nixpkgs/pull/288207)

wrong PR. sorry.

File: pkgs/development/python-modules/ibis-framework/default.nix [hunk 1]
```diff
@@ -17,23 +17,25 @@
 , geopandas
 , google-cloud-bigquery
 , google-cloud-bigquery-storage
-, graphviz-nox
+, graphviz
```
Comment (a-n-n-a-l-e-e) [inline pkgs/development/python-modules/ibis-framework/default.nix:20]:
```suggestion
, graphviz
```
why is this changed?

**Review by ghost — COMMENTED** [review]
Result of `nixpkgs-review pr 288207` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.ibis-framework</li>
    <li>python311Packages.ibis-framework.dist</li>
  </ul>
</details>

~except for using `graphviz` rather than the one without x, seems fine.~ [edit] `graphviz-nox` is not a python module so that change is needed.

**Comment by a-n-n-a-l-e-e** [comment]
> closes #289337

#289337 still has changes to the datafusion and some updates to the ibis unit tests which are worthwhile -- shouldn't be closed.

File: pkgs/development/python-modules/ibis-framework/default.nix (hunk 1 above)
Comment (cpcloud) [reply pkgs/development/python-modules/ibis-framework/default.nix:20]: Yeah, ideally this can be left as `graphviz-nox`. Ibis only uses graphviz to construct `dot` syntax, so no need to bring in X11

Comment (dotlambda) [reply pkgs/development/python-modules/ibis-framework/default.nix:20]: This is `python3Packages.graphviz`. We can make that package use `graphviz-nox` if you want.

**Review by ghost — APPROVED** [review]
errors building originated with the python packages which are listed as broken in https://malob.github.io/nix-review-tools-reports/nixpkgs:trunk/nixpkgs_trunk_1804408.html
- consonance
- tf2onnx

Result of `nixpkgs-review pr 288207` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages marked as broken and skipped:</summary>
  <ul>
    <li>apache-airflow</li>
    <li>apache-airflow.dist</li>
    <li>home-assistant-component-tests.roomba</li>
    <li>home-assistant-component-tests.snooz</li>
    <li>home-assistant-component-tests.sonos</li>
    <li>home-assistant-component-tests.volvooncall</li>
    <li>python312Packages.junos-eznc</li>
    <li>python312Packages.junos-eznc.dist</li>
    <li>python312Packages.pysnooz</li>
    <li>python312Packages.pysnooz.dist</li>
    <li>python312Packages.soco</li>
    <li>python312Packages.soco.dist</li>
    <li>python312Packages.streamz</li>
    <li>python312Packages.streamz.dist</li>
    <li>python312Packages.volvooncall</li>
    <li>python312Packages.volvooncall.dist</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>python311Packages.tf2onnx</li>
    <li>python311Packages.tf2onnx.dist</li>
    <li>python312Packages.consonance</li>
    <li>python312Packages.consonance.dist</li>
    <li>python312Packages.yowsup</li>
    <li>python312Packages.yowsup.dist</li>
  </ul>
</details>
<details>
  <summary>94 packages built:</summary>
  <ul>
    <li>apkleaks</li>
    <li>apkleaks.dist</li>
    <li>beets (beetsPackages.beets ,beetsPackages.beets-stable)</li>
    <li>beets-unstable (beetsPackages.beets-unstable)</li>
    <li>beets-unstable.dist (beetsPackages.beets-unstable.dist)</li>
    <li>beets-unstable.doc (beetsPackages.beets-unstable.doc)</li>
    <li>beets-unstable.man (beetsPackages.beets-unstable.man)</li>
    <li>beets.dist (beetsPackages.beets-stable.dist ,beetsPackages.beets.dist)</li>
    <li>beets.doc (beetsPackages.beets-stable.doc ,beetsPackages.beets.doc)</li>
    <li>beets.man (beetsPackages.beets-stable.man ,beetsPackages.beets.man)</li>
    <li>jadx</li>
    <li>mkchromecast</li>
    <li>mkchromecast.dist</li>
    <li>napalm</li>
    <li>napalm.dist</li>
    <li>nbutools</li>
    <li>nbutools.dist</li>
    <li>peering-manager</li>
    <li>python311Packages.amqtt</li>
    <li>python311Packages.amqtt.dist</li>
    <li>python311Packages.bambi</li>
    <li>python311Packages.bambi.dist</li>
    <li>python311Packages.catboost</li>
    <li>python311Packages.catboost.dist</li>
    <li>python311Packages.consonance</li>
    <li>python311Packages.consonance.dist</li>
    <li>python311Packages.diagrams</li>
    <li>python311Packages.diagrams.dist</li>
    <li>python311Packages.dissononce</li>
    <li>python311Packages.dissononce.dist</li>
    <li>python311Packages.graphviz</li>
    <li>python311Packages.graphviz.dist</li>
    <li>python311Packages.ibis-framework</li>
    <li>python311Packages.ibis-framework.dist</li>
    <li>python311Packages.insightface</li>
    <li>python311Packages.insightface.dist</li>
    <li>python311Packages.junos-eznc</li>
    <li>python311Packages.junos-eznc.dist</li>
    <li>python311Packages.mxnet</li>
    <li>python311Packages.mxnet.dist</li>
    <li>python311Packages.napalm</li>
    <li>python311Packages.napalm-hp-procurve</li>
    <li>python311Packages.napalm-hp-procurve.dist</li>
    <li>python311Packages.napalm.dist</li>
    <li>python311Packages.pipdeptree</li>
    <li>python311Packages.pipdeptree.dist</li>
    <li>python311Packages.pyhepmc</li>
    <li>python311Packages.pyhepmc.dist</li>
    <li>python311Packages.pysnooz</li>
    <li>python311Packages.pysnooz.dist</li>
    <li>python311Packages.pytm</li>
    <li>python311Packages.pytm.dist</li>
    <li>python311Packages.roombapy</li>
    <li>python311Packages.roombapy.dist</li>
    <li>python311Packages.skidl</li>
    <li>python311Packages.skidl.dist</li>
    <li>python311Packages.soco</li>
    <li>python311Packages.soco.dist</li>
    <li>python311Packages.streamz</li>
    <li>python311Packages.streamz.dist</li>
    <li>python311Packages.transitions</li>
    <li>python311Packages.transitions.dist</li>
    <li>python311Packages.volvooncall</li>
    <li>python311Packages.volvooncall.dist</li>
    <li>python311Packages.yowsup</li>
    <li>python311Packages.yowsup.dist</li>
    <li>python312Packages.amqtt</li>
    <li>python312Packages.amqtt.dist</li>
    <li>python312Packages.diagrams</li>
    <li>python312Packages.diagrams.dist</li>
    <li>python312Packages.dissononce</li>
    <li>python312Packages.dissononce.dist</li>
    <li>python312Packages.graphviz</li>
    <li>python312Packages.graphviz.dist</li>
    <li>python312Packages.mxnet</li>
    <li>python312Packages.mxnet.dist</li>
    <li>python312Packages.pyhepmc</li>
    <li>python312Packages.pyhepmc.dist</li>
    <li>python312Packages.roombapy</li>
    <li>python312Packages.roombapy.dist</li>
    <li>python312Packages.transitions</li>
    <li>python312Packages.transitions.dist</li>
    <li>quark-engine</li>
    <li>quark-engine.dist</li>
    <li>quodlibet-full</li>
    <li>quodlibet-full.dist</li>
    <li>quodlibet-full.doc</li>
    <li>quodlibet-xine-full</li>
    <li>quodlibet-xine-full.dist</li>
    <li>quodlibet-xine-full.doc</li>
    <li>sbomnix</li>
    <li>sbomnix.dist</li>
    <li>soco-cli</li>
    <li>soco-cli.dist</li>
  </ul>
</details>

---

errors building originated with the python packages which are listed as broken in https://malob.github.io/nix-review-tools-reports/nixpkgs:trunk/nixpkgs_trunk_1804408.html
- scikit-learn
- pyhepmc
- axolotl-curve25519
- transitions (not broken on hydra but flakey build, got tests to pass after a couple tries)

Result of `nixpkgs-review pr 288207` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>38 packages marked as broken and skipped:</summary>
  <ul>
    <li>apache-airflow</li>
    <li>apache-airflow.dist</li>
    <li>napalm</li>
    <li>napalm.dist</li>
    <li>python311Packages.napalm</li>
    <li>python311Packages.napalm-hp-procurve</li>
    <li>python311Packages.napalm-hp-procurve.dist</li>
    <li>python311Packages.napalm.dist</li>
    <li>python311Packages.pysnooz</li>
    <li>python311Packages.pysnooz.dist</li>
    <li>python311Packages.streamz</li>
    <li>python311Packages.streamz.dist</li>
    <li>python311Packages.tf2onnx</li>
    <li>python311Packages.tf2onnx.dist</li>
    <li>python312Packages.catboost</li>
    <li>python312Packages.catboost.dist</li>
    <li>python312Packages.insightface</li>
    <li>python312Packages.insightface.dist</li>
    <li>python312Packages.junos-eznc</li>
    <li>python312Packages.junos-eznc.dist</li>
    <li>python312Packages.pysnooz</li>
    <li>python312Packages.pysnooz.dist</li>
    <li>python312Packages.pytm</li>
    <li>python312Packages.pytm.dist</li>
    <li>python312Packages.soco</li>
    <li>python312Packages.soco.dist</li>
    <li>python312Packages.streamz</li>
    <li>python312Packages.streamz.dist</li>
    <li>python312Packages.tf2onnx</li>
    <li>python312Packages.tf2onnx.dist</li>
    <li>python312Packages.volvooncall</li>
    <li>python312Packages.volvooncall.dist</li>
    <li>quodlibet-full</li>
    <li>quodlibet-full.dist</li>
    <li>quodlibet-full.doc</li>
    <li>quodlibet-xine-full</li>
    <li>quodlibet-xine-full.dist</li>
    <li>quodlibet-xine-full.doc</li>
  </ul>
</details>
<details>
  <summary>34 packages failed to build:</summary>
  <ul>
    <li>python311Packages.amqtt</li>
    <li>python311Packages.amqtt.dist</li>
    <li>python311Packages.consonance</li>
    <li>python311Packages.consonance.dist</li>
    <li>python311Packages.dissononce</li>
    <li>python311Packages.dissononce.dist</li>
    <li>python311Packages.insightface</li>
    <li>python311Packages.insightface.dist</li>
    <li>python311Packages.junos-eznc</li>
    <li>python311Packages.junos-eznc.dist</li>
    <li>python311Packages.pyhepmc</li>
    <li>python311Packages.pyhepmc.dist</li>
    <li>python311Packages.roombapy</li>
    <li>python311Packages.roombapy.dist</li>
    <li>python311Packages.transitions</li>
    <li>python311Packages.transitions.dist</li>
    <li>python311Packages.volvooncall</li>
    <li>python311Packages.volvooncall.dist</li>
    <li>python311Packages.yowsup</li>
    <li>python311Packages.yowsup.dist</li>
    <li>python312Packages.amqtt</li>
    <li>python312Packages.amqtt.dist</li>
    <li>python312Packages.consonance</li>
    <li>python312Packages.consonance.dist</li>
    <li>python312Packages.dissononce</li>
    <li>python312Packages.dissononce.dist</li>
    <li>python312Packages.pyhepmc</li>
    <li>python312Packages.pyhepmc.dist</li>
    <li>python312Packages.roombapy</li>
    <li>python312Packages.roombapy.dist</li>
    <li>python312Packages.transitions</li>
    <li>python312Packages.transitions.dist</li>
    <li>python312Packages.yowsup</li>
    <li>python312Packages.yowsup.dist</li>
  </ul>
</details>
<details>
  <summary>39 packages built:</summary>
  <ul>
    <li>apkleaks</li>
    <li>apkleaks.dist</li>
    <li>jadx</li>
    <li>mkchromecast</li>
    <li>mkchromecast.dist</li>
    <li>nbutools</li>
    <li>nbutools.dist</li>
    <li>python311Packages.catboost</li>
    <li>python311Packages.catboost.dist</li>
    <li>python311Packages.diagrams</li>
    <li>python311Packages.diagrams.dist</li>
    <li>python311Packages.graphviz</li>
    <li>python311Packages.graphviz.dist</li>
    <li>python311Packages.ibis-framework</li>
    <li>python311Packages.ibis-framework.dist</li>
    <li>python311Packages.mxnet</li>
    <li>python311Packages.mxnet.dist</li>
    <li>python311Packages.pipdeptree</li>
    <li>python311Packages.pipdeptree.dist</li>
    <li>python311Packages.pytm</li>
    <li>python311Packages.pytm.dist</li>
    <li>python311Packages.skidl</li>
    <li>python311Packages.skidl.dist</li>
    <li>python311Packages.soco</li>
    <li>python311Packages.soco.dist</li>
    <li>python312Packages.diagrams</li>
    <li>python312Packages.diagrams.dist</li>
    <li>python312Packages.graphviz</li>
    <li>python312Packages.graphviz.dist</li>
    <li>python312Packages.mxnet</li>
    <li>python312Packages.mxnet.dist</li>
    <li>python312Packages.skidl</li>
    <li>python312Packages.skidl.dist</li>
    <li>quark-engine</li>
    <li>quark-engine.dist</li>
    <li>sbomnix</li>
    <li>sbomnix.dist</li>
    <li>soco-cli</li>
    <li>soco-cli.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/ibis-framework/default.nix [hunk 2]
```diff
@@ -94,25 +96,24 @@ buildPythonPackage rec {
...
     toolz
     typing-extensions
-  ]
-  ++ pooch.optional-dependencies.progress
-  ++ pooch.optional-dependencies.xxhash;
+  ];
 
   nativeCheckInputs = [
     pytestCheckHook
+    black
```
Comment (natsukium) [inline pkgs/development/python-modules/ibis-framework/default.nix:115]: Can we remove tests that depend on black?

**Review by ElleNajt — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/ibis-framework/default.nix (hunk 2 above)
Comment (dotlambda) [reply pkgs/development/python-modules/ibis-framework/default.nix:115]: Why? It seems easier to just depend on black.

**Comment by a-n-n-a-l-e-e** [comment]
> Maybe these are due to the change of graphviz, or maybe already broken on darwin, I'm not sure.

the packages that failed to build also were on x64 darwin, and were due to already broken packages.

File: pkgs/development/python-modules/ibis-framework/default.nix (hunk 2 above)
Comment (natsukium) [reply pkgs/development/python-modules/ibis-framework/default.nix:115]: In https://github.com/NixOS/nixpkgs/pull/274092#issuecomment-1858157156, @SuperSandro2000 said, 
> It reduces the amount of rebuilds triggered and by that potential breakages when bumping linters. It is nice that people want to keep high quality code but if we update e.g. black and a new rule breaks many things that is just plain annoying and unnecessary. Hence if the linters are not touching our code, the rule of thumb is to always remove them to keep maintenance required lower.

I agree with this opinion.

Comment (dotlambda) [reply pkgs/development/python-modules/ibis-framework/default.nix:115]: We don't test code quality here. Black is an optional dependency.
